typescript/cli/src/config/submit.ts

```diff
@@ -57,6 +62,29 @@ export async function runSubmit({
   }
 }
 
+/**
+ * Retrieves the chain name from transactions[0].
+ *
+ * @param multiProvider - The MultiProvider instance to use for chain name lookup.
+ * @param transactions - The list of populated transactions.
+ * @returns The name of the chain that the transactions are submitted on.
+ * @throws If the transactions are not all on the same chain or chain is not found
+ */
+function getChainFromTxs(
+  multiProvider: MultiProvider,
+  transactions: PopulatedTransactions,
+) {
+  const firstTransaction = transactions.at(0);
+  assert(firstTransaction, 'Transactions empty');
```

Comment:
maybe this can be part of the zod schema?

Reply:
yeah i think we can assume all PopulatedTransactions have the same chainId

Reply:
sorry I meant the non empty criteria